tests/tests/test-class-admin.php

```diff
@@ -427,29 +416,94 @@ public function test_get_users_record_meta() {
-		$this->admin->update_user_meta( get_current_user_id(), $key, $value );
-		$this->assertEquals( get_user_meta( get_current_user_id(), $key, true ), $value );
+		$user_id = $this->login_with_role( 'administrator' );
+		$key     = 'message_2';
+		$value   = 'I understand. It is important that you believe me. Look behind you.';
+		$this->admin->update_user_meta( $user_id, $key, $value );
+		$this->assertEquals( get_user_meta( $user_id, $key, true ), $value );
 	}
 
 	public function test_delete_user_meta() {
-		$key   = 'message_3';
-		$value = 'I was David Bowman.';
+		$user_id = $this->login_with_role( 'administrator' );
+		$key     = 'message_3';
+		$value   = 'I was David Bowman.';
+
+		update_user_meta( $user_id, $key, $value );
+		$this->assertEquals( get_user_meta( $user_id, $key, true ), $value );
+
+		$this->admin->delete_user_meta( $user_id, $key );
+
+		$this->assertEmpty( get_user_meta( $user_id, $key, true ) );
+	}
+
+	public function test_action_wp_stream_uninstall_registered() {
+		$this->assertTrue( has_action( 'wp_ajax_wp_stream_uninstall' ), 'Uninstall action registered' );
+	}
+
+	public function test_action_wp_stream_uninstall_success() {
+		$this->login_with_role( 'administrator' );
+		$_GET['nonce'] = wp_create_nonce( 'stream_uninstall_nonce' );
+
+		$db_driver_copy = $this->plugin->db->driver; // Save the original driver instance.
+
+		// Expect the purge_storage() call just once.
+		$db_driver_mock = $this->createMock( get_class( $this->plugin->db->driver ) );
+		$db_driver_mock->expects( $this->once() )->method( 'purge_storage' );
+		$this->plugin->db->driver = $db_driver_mock;
+
+		$this->_handleAjax( 'wp_stream_uninstall' );
+
+		$this->plugin->db->driver = $db_driver_copy; // Restore the original.
+	}
+
+	// Test that the action_wp_stream_uninstall() function requires a nonce when triggered via AJAX.
+	public function test_action_wp_stream_uninstall_without_nonce() {
+		$this->login_with_role( 'administrator' );
 
-		update_user_meta( get_current_user_id(), $key, $value );
-		$this->assertEquals( get_user_meta( get_current_user_id(), $key, true ), $value );
+		$this->expectException( 'WPAjaxDieStopException' );
+		$this->expectExceptionMessage( '-1' );
+
+		$this->_handleAjax( 'wp_stream_uninstall' );
+	}
+
+	// Test that the action_wp_stream_uninstall() function can only be called with admin capabilities.
+	public function test_action_wp_stream_uninstall_requires_admin_caps() {
+		$_GET['nonce'] = wp_create_nonce( 'stream_uninstall_nonce' );
 
-		$this->admin->delete_user_meta( get_current_user_id(), $key );
+		$this->expectException( 'WPAjaxDieStopException' );
+		$this->expectExceptionMessage( 'You don&#039;t have sufficient privileges to do this action.' );
 
-		$this->assertEmpty( get_user_meta( get_current_user_id(), $key, true ) );
+		$this->_handleAjax( 'wp_stream_uninstall' );
+	}
+
+	public function test_action_wp_stream_uninstall_not_available_to_non_admins() {
```

Comment:
All the tests related to the `stream_uninstall_nonce` action are now part of the admin test suite since that is where the controller logic for this lives.